R/Phenotyping.R

```diff
@@ -374,55 +374,63 @@ ExpandGeneList <- function(genes, verbose = TRUE) {
 
 #' @title MakePhenotypingDotPlot
 #'
-#' @description A helper function that Takes an input gene and identifies any entries matching registered gene sets. This function is the complement to GetGeneSet.
+#' @description Creates a DotPlot using custom gene sets and attempts to coarsely group gene sets by cell type. 
 #' @param yField The grouping variable used to calculate the average expression of genes and the y axis of the DotPlot.
-#' @param colorField Whether to color dots by scaled expression or unscaled expression.
+#' @param scaled A boolean defining whether to color dots by scaled expression or unscaled expression.
+#' @param gene_lists A vector of gene lists (defined by .RegisterGeneSet) to be queried and their genes be plotted.
 MakePhenotypingDotPlot <- function(seuratObj,
                                    yField = 'ClusterNames_0.2',
-                                   colorField = 'scaled'
+                                   scaled = T, 
+                                   gene_lists = c('Cytotoxicity', 'EffectorCytokines')
 ){
-  #Establish which gene sets will be queried  
-  meta_gene_list <- c(RIRA::GetGeneSet('TandNK_Activation.1'), RIRA::GetGeneSet('MemoryAndNaive'), RIRA::GetGeneSet('Cytotoxicity'), RIRA::GetGeneSet('ExhaustionOrInhibitory'), RIRA::GetGeneSet('Myelocytes'), RIRA::GetGeneSet('Pro_Myelocytes'))
+  if (!is.logical(scaled)){
+    warning("Please ensure scaled is either TRUE (to use per-gene scaled expression) or FALSE (for raw gene expression)")
```

Comment:
I think you can just do:

stop("please ensure....")

instead of a warning and an empty stop()